packages/foundry/test/BalancerFun.t.sol

```diff
@@ -0,0 +1,152 @@
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+
+import { BaseVaultTest } from "@balancer-labs/v3-vault/test/foundry/utils/BaseVaultTest.sol";
+import { PoolMock } from "@balancer-labs/v3-vault/contracts/test/PoolMock.sol";
+
+import { BalancerFun } from "../contracts/hooks/BalancerFun.sol";
+
+/**
+ * @title BalancerFunTest
+ * @notice Unit tests for the BalancerFun contract.
+ * @dev Inherits from BaseVaultTest to perform setup and test BalancerFun interactions.
+ */
+contract BalancerFunTest is BaseVaultTest {
+    using CastingHelpers for address[];
+    using FixedPoint for uint;
+
+    uint internal daiIdx;
+    uint internal usdcIdx;
+
+    /**
+     * @notice Sets up the test environment.
+     * @dev Overrides BaseVaultTest's setUp function to initialize token indexes.
+     */
+    function setUp() public virtual override {
+        BaseVaultTest.setUp();
+        (daiIdx, usdcIdx) = getSortedIndexes(address(dai), address(usdc));
+    }
+
+    /**
+     * @notice Creates a new BalancerFun hook for testing.
+     * @dev Deploys a new instance of BalancerFun and sets it as the hook for the pool.
+     * @return address The address of the newly created hook.
+     */
+    function createHook() internal override returns (address) {
+        // lp will be the owner of the hook. Only the owner can set hook fee percentages.
+        vm.prank(lp);
+        BalancerFun hook = new BalancerFun(IVault(address(vault)), IERC20(address(router)));
+        return address(hook);
+    }
+
+    /**
+     * @notice Creates a new pool with custom liquidity management settings.
+     * @dev Overrides the pool creation to disable unbalanced liquidity by setting liquidityManagement.
+     * @param tokens The tokens to be used in the pool.
+     * @param label A label for the pool.
+     * @return address The address of the newly created pool.
+     */
+    function _createPool(address[] memory tokens, string memory label) internal override returns (address) {
+        PoolMock newPool = new PoolMock(IVault(address(vault)), "Balancer.Fun Pool", "BALFUN");
+        vm.label(address(newPool), label);
+        PoolRoleAccounts memory roleAccounts;
+        roleAccounts.poolCreator = lp;
+        LiquidityManagement memory liquidityManagement;
+        factoryMock.registerPool(
+            address(newPool),
+            vault.buildTokenConfig(tokens.asIERC20()),
+            roleAccounts,
+            poolHooksContract,
+            liquidityManagement
+        );
+
+        return address(newPool);
+    }
+
+    /**
+     * @notice Tests the setup of the contract.
+     * @dev Verifies that the setup has been completed successfully.
+     */
+    function testSetUp() public {
+        assertEq(daiIdx, 0, "SetUp has failed");
+    }
+    
+    /**
+     * @notice Tests a swap operation.
+     * @dev Executes a swap and verifies the balance changes for Alice.
+     */
+    function testSwap() public {
```

Comment:
It would be nice to see a test that actually triggers one of the new conditions errors e.g. `MaximumSwapExceeded`